src/utils/documentdb_model.ts

```diff
@@ -47,51 +83,53 @@ export abstract class DocumentDbModel<
    *     resolves to either an error or the retrieved version of the persisted
    *     document.
    */
-  public async create(
-    document: TN,
-    partitionKey: string
-  ): Promise<Either<DocumentDb.QueryError, TR>> {
-    // we don't want to store the "kind" property, let's remove it
-    const kindlessDocument = Object.assign(Object.assign({}, document), {
-      kind: undefined
-    });
-
-    // attempt to persist the document
-    const maybeCreatedDocument = await DocumentDbUtils.createDocument(
-      this.dbClient,
-      this.collectionUri,
-      kindlessDocument,
-      partitionKey
-    );
-
-    // if the result is successful we map it to a TR type
-    return maybeCreatedDocument.map(this.toRetrieved);
+  public create(
+    document: T,
+    options?: RequestOptions
+  ): TaskEither<QErrors, t.TypeOf<TR<T>>> {
+    const createItem = this.container.items.create;
+    const item = this.type.encode(document);
+    return tryCatch<QErrors, PromiseType<ReturnType<typeof createItem>>>(
+      () => createItem(item, options),
+      e => ({
+        error: toError(e),
+        kind: "RESPONSE"
+      })
+    )
+      .map(_ => _.resource)
+      .filterOrElse(isDefined, { kind: "EMPTY" }) // TODO: not sure whether an undef resource should be an error
+      .chain(_ =>
+        fromEither(
+          this.typeR.decode(_).mapLeft(error => ({
+            error,
+            kind: "VALIDATION"
+          }))
+        )
+      );
   }
 
   /**
    * Creates a new document or update an existing one
    * with the provided id.
    *
    */
-  public async createOrUpdate(
+  public createOrUpdate(
     document: TN,
-    partitionKey: string
-  ): Promise<Either<DocumentDb.QueryError, TR>> {
+    options?: RequestOptions
+  ): TaskEither<ErrorResponse, TR> {
```

Comment:
Why ErrorResponse here instead of QErrors?

Reply:
let me know what do you think about how's `create` structured